### ISSUE TITLE
Adding new branch using the "Add a branch" popover not reflected in the code editor

### ISSUE THREAD

**Issue by tplevko** [KaotoIO/kaoto-ui#1381]
### Describe the Bug

When user adds new branch for the Choice step using the "Add a branch" popover -   it adds a otherwise step:
```
otherwise:
      steps: []
```
This "otherwise" step can be added multiple times. Also the new branch is not reflected in the code editor.

### Steps to Reproduce the Bug or Issue

1. Add a branch to choice step using the "Add a branch" popover
2. It's not visible in the editor

### Screenshots or Videos

![Peek 2023-03-03 14-43](https://user-images.githubusercontent.com/4180208/222737007-2765a949-82ab-41ae-9751-27a6ab6c15e7.gif)

### Platform

- OS: [Linux]
- Browser: [Firefox]
- Version: [latest]

**Comment by kahboom** [comment]
Uh oh that's not good..

**Comment by kahboom** [comment]
On hold due to https://github.com/KaotoIO/kaoto-backend/issues/512

**Comment by lordrip** [comment]
I was looking into this issue, so given that it was blocked by [another issue](https://github.com/KaotoIO/kaoto-backend/issues/512), I put together this [draft pull request](https://github.com/KaotoIO/kaoto-backend/pull/518) to discuss what to do in those cases

**Comment by lordrip** [comment]
now that the previous pull request it's merged, this issue is no longer on hold 🚀

**Comment by lordrip** [comment]
@Delawen I pretty-please summon your wisdom here :)

Let's consider the following case:

1. I start with a `kamelet:source` step, so we send this indivisual step to the backend to receive the coerresponding `yaml`
2. Then we add a `choice` step and we send now to the backend 2 steps and with this last step `{ branches: null }`
```
{
  dsl: "Camel Route"
  metadata: {
    description: ""
    dsl: { ... }
  },
  steps: [
    { ... kamelet:source ... },
    {
      branches: null
      id: "choice-MIDDLE"
      kind: "EIP-BRANCH"
      maxBranches: -1
      minBranches: 1
      name: "choice"
    }
  ]
}
```
3. We receive back:
```yaml
- from:
    uri: kamelet:source
    steps:
    - choice:
        when: [] # <--- this seems automatically generated
```
4. After that, we add a new branch and then we send to the backend the same payload but now including the newly added branch:
```
branchUuid: "b-1-670523182"
identifier: ""
steps: []
```
5. The response is:
```yaml
- from:
    uri: kamelet:source
    steps:
    - choice:
        when: []
        otherwise: # <--- :eyes: how this otherwise came here?
          steps: []
```

Now the question is, is this the right behavior or are we missing some information that we should send to the backend to properly attach branches with their UI counterpart :thinking: ?

**Comment by Delawen** [comment]
That otherwise appears because you didn't add any conditional to the branch you sent. Meaning it is not a `when` but an `otherwise`

**Comment by Delawen** [comment]
That won't happen if we use the step extension to create branches.

**Comment by lordrip** [comment]
> That won't happen if we use the step extension to create branches.

I see, probably without having the knowledge of the step details in the UI at the moment, it's somewhat confusing to be able to add branches from the mini catalog but then don't see it directly in the code editor :thinking: 
![image](https://user-images.githubusercontent.com/16512618/223961792-ee80dd3c-cc87-4158-a9ff-d5d0410fff7d.png)


Alrighty, that being said, how do we feel about the following:

_In case the step has an extension registered, we keep the `Branches` tab and we replace the `Add a Branch` button with an info message inviting the user to click on the step instead._
![Screenshot from 2023-03-08 13-49-07](https://user-images.githubusercontent.com/16512618/223962499-1c3d30f3-0737-4156-882d-e0f60e653cf7.png)

@Mdenisco do you have some tips here? :), I was also thinking if would be helpful to add a gif image showing a mouse pointer clicking on a step to be extra clear :thinking: 

_P.S. Please don't mind the text at the moment :sweat_smile:_

**Comment by kahboom** [comment]
@lordrip I'm not a big fan of keeping the tab just for a text hint (though I can understand a similar reasoning for just an "Add branch" button), but if it will be replaced by some extension then it makes sense. Still, I also have concerns about requiring extensions for these kind of things. Since all of this is just hypothetical discussions, maybe let's go with this solution and we can re-evaluate after the release.

**Comment by lordrip** [comment]
@kahboom @mmelko I hear you :raised_hands:, hopefully this is better

![image](https://user-images.githubusercontent.com/16512618/224360671-a1888198-6647-4e7c-becf-230f21d3ff1d.png)
